src/main/java/frc/utils/ToleranceUtils.java

```diff
@@ -0,0 +1,28 @@
+package frc.utils;
+
+
+import edu.wpi.first.math.MathUtil;
+import edu.wpi.first.math.geometry.Rotation2d;
+import frc.constants.MathConstants;
+
+public class ToleranceUtils {
+
+	public static boolean isNearWrapped(Rotation2d wantedAngle, Rotation2d angle, Rotation2d tolerance) {
+		return MathUtil.isNear(
+			wantedAngle.getRadians(),
+			angle.getRadians(),
+			tolerance.getRadians(),
+			MathConstants.HALF_CIRCLE.unaryMinus().getRadians(),
```

Comment:
if rotation2d wraps it in modulo automatically it won't work. 
Have you done static tests yet?